org.civicrm.reqangular/src/common/angular-date/src/services/DateValidationService.js

```diff
@@ -56,7 +56,7 @@ function DateValidationService($filter) {
 
     me.validate = function validate(start_date, end_date) {
```

Comment:
```javascript
me._error = function (error_msg, fields) {
    throw new Error(error_msg);
};
```
'fields' param, although isn't used within default error function, provides information on which date field error was thrown. i.e:
```javascript
me._error('Date is invalid! Valid date format is: dd/mm/yyyy.', ['end_date']);
```
Developer can override this function, and use fields parameter. Basically this param is here, because error function takes 2 parameters always, even if there is no purpose in using one of them. (for example we are validating one datefield - we don't need information about field).

It gives freedom do provide custom error functions as done in HRJobRoles:
```javascript
 $scope.validateDates(data.start_date.$viewValue, data.end_date.$viewValue, function(error, field){
                    errors++;
                    if(field.indexOf('start_date') > -1) {
                        data.start_date.$error.custom.push(error);
                    }
                    if(field.indexOf('end_date') > -1){
                        data.end_date.$error.custom.push(error);
                    }
                });
```